ketcher-autotests/tests/Structure-Creating-&-Editing/Text-Tool/text-tools-modification-and-manipulation.spec.ts

```diff
@@ -0,0 +1,311 @@
+import { Page, test } from '@playwright/test';
+import {
+  takeEditorScreenshot,
+  selectLeftPanelButton,
+  selectTopPanelButton,
+  waitForPageInit,
+  pressButton,
+  LeftPanelButton,
+  TopPanelButton,
+  openFileAndAddToCanvas,
+  clickInTheMiddleOfTheScreen,
+  dragMouseTo,
+  selectRing,
+  RingButton,
+  selectDropdownTool,
+  waitForRender,
+} from '@utils';
+import { addTextBoxToCanvas } from '@utils/selectors/addTextBoxToCanvas';
+
+async function createSomeStructure(page: Page) {
+  const a = 97;
+  const b = 79;
+  const c = 943;
+  const d = 114;
+  const e = 844;
+  const f = 579;
+  const g = 66;
+  const h = 611;
+  await page.mouse.move(a, b);
+  await page.mouse.down();
+  await page.mouse.move(c, d);
+  await page.mouse.move(e, f);
+  await page.mouse.move(g, h);
+  await page.mouse.up();
+}
+
+async function createSomeMovement(page: Page) {
+  const i = 656;
+  const j = 359;
+  const k = 906;
+  const l = 245;
+  await page.mouse.move(i, j);
+  await page.mouse.down();
+  await dragMouseTo(k, l, page);
+  await page.mouse.up();
+}
+
+test.describe('Text tools test cases', () => {
+  test.beforeEach(async ({ page }) => {
+    await waitForPageInit(page);
+  });
+
+  test.afterEach(async ({ page }) => {
+    await takeEditorScreenshot(page);
+  });
+
+  test('Text tool - Modify the created text object', async ({ page }) => {
+    // Test case: EPMLSOPKET-2228
+    // Checking if possible is modify created text object by adding some extra symbols
+    await addTextBoxToCanvas(page);
+    await page.getByRole('dialog').getByRole('textbox').fill('TEST');
+    await pressButton(page, 'Apply');
+    await takeEditorScreenshot(page);
+    await page.getByText('TEST').dblclick();
+    await page.getByRole('dialog').getByRole('textbox').click();
+    await page.getByRole('dialog').getByRole('textbox').fill('TEST123');
+    await pressButton(page, 'Apply');
+  });
+
+  test('Text tool - Delete the created text object', async ({ page }) => {
+    // Test case: EPMLSOPKET-2229
+    // Delte created text object with Erase tool
+    await addTextBoxToCanvas(page);
+    await page.getByRole('dialog').getByRole('textbox').fill('TEST');
+    await pressButton(page, 'Apply');
+    await selectLeftPanelButton(LeftPanelButton.Erase, page);
+    await page.getByText('TEST').click();
+    await waitForRender(page, async () => {
+      await selectTopPanelButton(TopPanelButton.Undo, page);
+      await selectTopPanelButton(TopPanelButton.Redo, page);
+      await selectTopPanelButton(TopPanelButton.Undo, page);
```

Comment:
This code is repeated often. I suggest creating a function for example
async function performUndoRedo(page: Page) {
   await waitForRender(page, async () => {
      await selectTopPanelButton(TopPanelButton.Undo, page);
      await selectTopPanelButton(TopPanelButton.Redo, page);
      await selectTopPanelButton(TopPanelButton.Undo, page);
}